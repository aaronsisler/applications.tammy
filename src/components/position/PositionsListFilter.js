import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { setPositionsTextFilter } from 'Actions/filters/positionsText';

export class PositionsListFilter extends React.Component {
    onTextChange = (e) => {
        this.props.setPositionsTextFilter(e.target.value);
    }

    render() {
        return (
            <div className="positions_list_filters">
                <div className="input_group__item">
                    <input
                        type="text"
                        className="text_input"
                        placeholder="Search positions"
                        value={this.props.filters.text}
                        onChange={this.onTextChange}
                    />
                </div>
            </div>
        );
    }
}

/* istanbul ignore next */
const mapStateToProps = (state) => ({
    filters: state.filters.positions
});

/* istanbul ignore next */
const mapDispatchToProps = (dispatch) => ({
    setPositionsTextFilter: (text) => dispatch(setPositionsTextFilter(text)),
});

export default connect(mapStateToProps, mapDispatchToProps)(PositionsListFilter);

PositionsListFilter.propTypes = {
    filters: PropTypes.object.isRequired,
    setPositionsTextFilter: PropTypes.func.isRequired,
};

