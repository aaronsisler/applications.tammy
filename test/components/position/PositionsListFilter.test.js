import React from 'react';
import { shallow } from 'enzyme';
import { PositionsListFilter } from 'Position/PositionsListFilter';

describe('PositionsListFilter', () => {
    const setPositionsTextFilter = jest.fn();
    const filters = { text: 'initial filter' };
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(
            <PositionsListFilter
                filters={filters}
                setPositionsTextFilter={setPositionsTextFilter}
            />
        );
    })

    it('should render PositionsListFilter correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should call setPositionsTextFilter prop on text input change', () => {
        const value = 'added text';
        const eventMock = { target: { value } };

        wrapper.find('input').simulate('change', eventMock);

        expect(setPositionsTextFilter).toHaveBeenLastCalledWith(value);
    });
});
