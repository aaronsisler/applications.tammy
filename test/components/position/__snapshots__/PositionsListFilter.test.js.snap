// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PositionsListFilter should render PositionsListFilter correctly 1`] = `
<div
  className="positions_list_filters"
>
  <div
    className="input_group__item"
  >
    <input
      className="text_input"
      onChange={[Function]}
      placeholder="Search positions"
      type="text"
      value="initial filter"
    />
  </div>
</div>
`;
